kasa/smartdevice.py

```diff
@@ -758,3 +752,42 @@ def __repr__(self):
             f" ({self.alias}), is_on: {self.is_on}"
             f" - dev specific: {self.state_information}>"
         )
+
+    @staticmethod
+    async def connect(
+        host: str,
+        *,
+        port: Optional[int] = None,
+        timeout=5,
+        credentials: Optional[Credentials] = None,
+        device_type: Optional[DeviceType] = None,
+    ) -> "SmartDevice":
+        """Connect to a single device by the given IP address.
+
+        This method avoids the UDP based discovery process and
+        will connect directly to the device to query its type.
+
+        It is generally preferred to avoid :func:`discover_single()` and
+        use this function instead as it should perform better when
+        the WiFi network is congested or the device is not responding
+        to discovery requests.
+
+        The device type is discovered by querying the device.
+
+        :param host: Hostname of device to query
+        :param device_type: Device type to use for the device.
+            If not given, the device type is discovered by querying the device.
+            If the device type is already known, it is preferred to pass it
+            to avoid the extra query to the device to discover its type.
+        :rtype: SmartDevice
+        :return: Object for querying/controlling found device.
+        """
+        from .device_factory import connect  # pylint: disable=import-outside-toplevel
```

Comment:
The downside to putting it here is we have to do a late import otherwise we have a circular dep

Reply:
I think this is fine.